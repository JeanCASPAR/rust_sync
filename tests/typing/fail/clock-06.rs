use rustre::sync;

sync! {
    #![pass(1)]

    node oui(c : bool) = ()
    where
        a : float = merge c {
            true => 3.0,
            false => 2.0,
        };
}

fn main() {}
