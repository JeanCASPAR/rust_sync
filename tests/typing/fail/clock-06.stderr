error: clock error: the expression of the true branch in a merge has the base clock
 --> tests/typing/fail/clock-06.rs:9:21
  |
9 |             true => 3.0,
  |                     ^^^
  |
  = help: Try adding ` when c`.
